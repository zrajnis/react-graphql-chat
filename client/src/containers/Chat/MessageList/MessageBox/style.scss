.my-message-wrapper,
.chat-message-wrapper {
  .chat-message {
    display: inline-block;
    font-size: 18px;
    margin: 15px 0;
    max-width: 600px;
    padding: 15px;
    width: inherit;

    p {
      display: inline-block;
      margin: 0 0 0 10px;
    }
  }
}

.chat-message-wrapper {
  .chat-message {
    background: $bg-white;
    border-radius: 0 10px 10px;
    color: $black;
  }
}

.my-message-wrapper {
  text-align: right;

  .chat-message {
    background: $blue;
    border-radius: 10px 0 10px 10px;
    color: $white;
  }
}
