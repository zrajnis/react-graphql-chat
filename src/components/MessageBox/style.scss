.my-message-wrapper,
.chat-message-wrapper {
  .chat-message {
    display: inline-block;
    font-size: 18px;
    margin: 15px 0;
    padding: 15px;
    width: inherit;
    max-width: 600px;

    p {
      display: inline-block;
      margin: 0 0 0 10px;
    }
  }
}

.chat-message-wrapper {
  .chat-message {
    background: $bgWhite;
    border-radius: 0 10px 10px 10px;
    color: $black;
  }
}

.my-message-wrapper {
  text-align: right;
  .chat-message {
    background: $blue;
    border-radius: 10px 0 10px 10px;
    color: $white;
  }
}
