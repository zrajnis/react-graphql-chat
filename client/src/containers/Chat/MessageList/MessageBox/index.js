import React from 'react'
import PropTypes from 'prop-types'

import './style.scss'

const MessageBox = ({ message, myMessage }) => (
  <li styleName={myMessage ? 'my-message-wrapper' : 'chat-message-wrapper'}>
    <div styleName='chat-message'>
      {!myMessage && <span>{message.from}:</span>}
      <p>
        {message.content}
      </p>
    </div>
  </li>
)

MessageBox.propTypes = {
  message: PropTypes.shape({
    content: PropTypes.string.isRequired,
    from: PropTypes.string.isRequired
  }).isRequired,
  myMessage: PropTypes.bool.isRequired
}

export default MessageBox
