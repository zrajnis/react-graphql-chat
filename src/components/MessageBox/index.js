import React from 'react'
import PropTypes from 'prop-types'

import './style.scss'

const MessageBox = ({ message, myMessage }) => (
  <div styleName={myMessage ? 'my-message-wrapper' : 'chat-message-wrapper'}>
    <div styleName='chat-message'>
      {!myMessage && <span>{message.from}:</span>}
      <p>
        {message.content}
      </p>
    </div>
  </div>
)

MessageBox.propTypes = {
  message: PropTypes.shape({
    content: PropTypes.string.isRequired,
    from: PropTypes.string.isRequired
  }).isRequired,
  myMessage: PropTypes.bool.isRequired
}

export default MessageBox
