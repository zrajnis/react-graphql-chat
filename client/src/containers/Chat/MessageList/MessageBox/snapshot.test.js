import React from 'react'
import renderer from 'react-test-renderer'

import MessageBox from 'containers/Chat/MessageList/MessageBox'

describe('Message box component snapshot', () => {
  const message = {
    content: 'test content 123',
    from: 'testUser'
  }

  it('renders the message box from logged in user', () => {
    const tree = renderer.create(
      <MessageBox
        message={message}
        myMessage
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders the message box from other users', () => {
    const tree = renderer.create(
      <MessageBox
        message={message}
        myMessage={false}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
