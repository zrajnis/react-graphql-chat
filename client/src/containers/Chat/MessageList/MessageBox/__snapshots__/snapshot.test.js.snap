// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message box component snapshot renders the message box from logged in user 1`] = `
<li
  className="my-message-wrapper--h93b8ATj"
>
  <div
    className="chat-message--QnC2_lx7"
  >
    <p>
      test content 123
    </p>
  </div>
</li>
`;

exports[`Message box component snapshot renders the message box from other users 1`] = `
<li
  className="chat-message-wrapper--hp6HY0Bp"
>
  <div
    className="chat-message--QnC2_lx7"
  >
    <span>
      testUser
      :
    </span>
    <p>
      test content 123
    </p>
  </div>
</li>
`;
